src/cmfd_execute.F90

```diff
@@ -276,10 +271,21 @@ subroutine cmfd_reweight(new_weights)
 
       ! Have master compute weight factors (watch for 0s)
       if (master) then
-        where(cmfd % cmfd_src > ZERO .and. cmfd % sourcecounts > ZERO)
-          cmfd % weightfactors = cmfd % cmfd_src/sum(cmfd % cmfd_src)* &
-                               sum(cmfd % sourcecounts) / cmfd % sourcecounts
-        end where
+        ! Calculate normalization factor
+        norm = sum(cmfd % sourcecounts) / sum(cmfd % cmfd_src)
+
+        do mesh_bin = 1, nx*ny*nz
+          call cmfd_mesh % get_indices_from_bin(mesh_bin, ijk)
+          do g = 1, ng
+            if (cmfd % sourcecounts(ng - g + 1, mesh_bin) > ZERO) then
+              if (cmfd % cmfd_src(g,ijk(1),ijk(2),ijk(3)) > ZERO) then
+                cmfd % weightfactors(g,ijk(1),ijk(2),ijk(3)) = &
+                     cmfd % cmfd_src(g,ijk(1),ijk(2),ijk(3)) * norm &
+                     / cmfd % sourcecounts(ng - g + 1, mesh_bin)
```

Comment:
@paulromano - what was the motivation for changing this, just curious?

Reply:
I ended up changing this when I was changing how `count_bank_sites` worked. Before, as you'll recall, it operated on a 4D array where the dimension corresponded to an energy grid and the x- y- and z- dimensions of a mesh. I wanted it to work instead on a 2D array of (energy, mesh bin) so that it potentially could work with any type of mesh. That caused `cmfd % sourcecounts` to become a 2D array instead of 4D. As a result, it wasn't possible to use the `WHERE` block to calculate `cmfd % cmfd_src` because it was still 4D whereas `sourcecounts` was 2D, so it got changed to nested do loops.